calico/felix/fetcd.py

```diff
@@ -207,6 +212,38 @@ def _periodically_report_status(self):
                 sleep_time = interval + jitter
                 gevent.sleep(sleep_time)
 
+    @logging_exceptions
+    def _periodically_version_signal(self):
+        """
+        Greenlet: periodically check Project Calico for version info
+
+        :return: Does not return, unless reporting disabled.
+        """
+        _log.info("Started version signal thread. Waiting for config.")
+        self._watcher.configured.wait()
```

Comment:
If you moved this to a top-level greenlet and started it after the config is loaded then you wouldn't need to block on this event.  (By the time we start the prometheus thread from `_main_greenlet()`, the config is loaded and we've already waited on this event.